tests/parity/test_catalog.py

```
@@ -0,0 +1,197 @@
+@pytest.mark.parametrize("source_pack", ["v2_non_soak", "agent_sdk_boundary"])
+@pytest.mark.parametrize("path_name", ["positive_path"])
+def test_catalog_rejects_nonrequired_positive_path_for_partial_source_packs(
+    tmp_path, source_pack, path_name
+) -> None:
+    document = yaml.safe_load(CATALOG_PATH.read_text(encoding="utf-8"))
+    row = next(item for item in document["capabilities"] if item["source_pack"] == source_pack)
+    row[path_name]["required"] = False
+    row[path_name]["rationale"] = "tampered applicability"
+    path = tmp_path / "nonrequired-positive.yaml"
+    path.write_text(yaml.safe_dump(document, sort_keys=False), encoding="utf-8")
+    with pytest.raises(CatalogViolation, match="positive_path.required must be true"):
```

Ruff (0.16.3)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)